node/src/accountant/scanners/test_utils.rs

```diff
@@ -2,9 +2,95 @@
 
 #![cfg(test)]
 
-use crate::accountant::db_access_objects::payable_dao::PayableAccount;
-use masq_lib::type_obfuscation::Obfuscated;
+use crate::accountant::scanners::local_test_utils::ScannerMock;
+use crate::accountant::scanners::payable_scanner_extension::msgs::QualifiedPayablesMessage;
+use crate::accountant::scanners::scan_schedulers::NewPayableScanDynIntervalComputer;
+use crate::accountant::scanners::scanners_utils::payable_scanner_utils::PayableScanResult;
+use crate::accountant::scanners::scanners_utils::pending_payable_scanner_utils::PendingPayableScanResult;
+use crate::accountant::scanners::{PayableScanner, PendingPayableScanner, ReceivableScanner};
+use crate::accountant::{
+    ReceivedPayments, ReportTransactionReceipts, RequestTransactionReceipts, SentPayables,
+};
+use crate::blockchain::blockchain_bridge::RetrieveTransactions;
+use masq_lib::logger::Logger;
+use masq_lib::ui_gateway::NodeToUiMessage;
+use std::cell::RefCell;
+use std::sync::{Arc, Mutex};
+use std::time::{Duration, SystemTime};
 
-pub fn protect_payables_in_test(payables: Vec<PayableAccount>) -> Obfuscated {
-    Obfuscated::obfuscate_vector(payables)
+#[derive(Default)]
+pub struct NewPayableScanDynIntervalComputerMock {
+    compute_interval_params: Arc<Mutex<Vec<(SystemTime, SystemTime, Duration)>>>,
+    compute_interval_results: RefCell<Vec<Option<Duration>>>,
+}
+
+impl NewPayableScanDynIntervalComputer for NewPayableScanDynIntervalComputerMock {
+    fn compute_interval(
+        &self,
+        now: SystemTime,
+        last_new_payable_scan_timestamp: SystemTime,
+        interval: Duration,
+    ) -> Option<Duration> {
+        self.compute_interval_params.lock().unwrap().push((
+            now,
+            last_new_payable_scan_timestamp,
+            interval,
+        ));
+        self.compute_interval_results.borrow_mut().remove(0)
+    }
+}
+
+impl NewPayableScanDynIntervalComputerMock {
+    pub fn compute_interval_params(
+        mut self,
+        params: &Arc<Mutex<Vec<(SystemTime, SystemTime, Duration)>>>,
+    ) -> Self {
+        self.compute_interval_params = params.clone();
+        self
+    }
+
+    pub fn compute_interval_result(self, result: Option<Duration>) -> Self {
+        self.compute_interval_results.borrow_mut().push(result);
+        self
+    }
+}
+
+pub enum ReplacementType<A, B> {
+    Real(A),
+    Mock(B),
+    Null,
+}
+
+// The supplied scanner types are broken down to these detailed categories because they are
+// eventually represented by a private trait within the Scanners struct. Therefore, when
+// the values are constructed, they cannot be made into a trait object right away and needs to be
+// handled specifically.
```

Comment:
You'd like to reword there...

because they become eventually trait objects represented by a private trait. That one cannot be supplied, though, because it is unaccessable from outside, therefore, we provide the specific objects that will be then treated in an abstract manner.